### PR TITLE
Implement CreateGroup

### PR DESCRIPTION
**Pull request by bduffany**

### CODE REVIEW

File: server/buildbuddy_server/buildbuddy_server.go [hunk 1]
```diff
@@ -117,7 +117,42 @@ func (s *BuildBuddyServer) CreateUser(ctx context.Context, req *uspb.CreateUserR
...
+	userDB := s.env.GetUserDB()
+	if auth == nil || userDB == nil {
+		return nil, status.UnimplementedError("Not Implemented")
+	}
+	jwtUser := &tables.User{}
+	if err := auth.FillUser(ctx, jwtUser); err != nil {
+		return nil, err
+	}
+
+	groupName := strings.TrimSpace(req.Name)
```
Comment (tylerwilliams) [inline server/buildbuddy_server/buildbuddy_server.go:130]: generally nicer to use the "Get*" accessors on proto fields so you know you're not getting a nil back.

Can you switch to GetName here and GetAutoPopulateFromOwnedDomain below?

File: server/buildbuddy_server/buildbuddy_server.go [hunk 2]
```diff
@@ -117,7 +117,42 @@ func (s *BuildBuddyServer) CreateUser(ctx context.Context, req *uspb.CreateUserR
...
+	if len(groupName) == 0 {
+		return nil, status.InvalidArgumentError("Group name cannot be empty")
+	}
+
+	var groupOwnedDomain string = ""
```
Comment (tylerwilliams) [inline server/buildbuddy_server/buildbuddy_server.go:135]: this is fine, but if you want to be more concise you can do:
groupOwnedDomain := ""

Typically the "var x type" idiom is used when you  have some conditional assignment happening inside a scope or similar.

**Review by tylerwilliams — CHANGES_REQUESTED** [review]

File: server/buildbuddy_server/buildbuddy_server.go (hunk 1 above)
Comment (bduffany) [reply server/buildbuddy_server/buildbuddy_server.go]: Thanks for the tip! Done

File: server/buildbuddy_server/buildbuddy_server.go (hunk 2 above)
Comment (bduffany) [reply server/buildbuddy_server/buildbuddy_server.go]: Done

**Review by tylerwilliams — APPROVED** [review]
LGTM

File: server/buildbuddy_server/buildbuddy_server.go [hunk 3]
```diff
@@ -117,7 +117,42 @@ func (s *BuildBuddyServer) CreateUser(ctx context.Context, req *uspb.CreateUserR
...
+	group := &tables.Group{
+		UserID:      jwtUser.UserID,
+		Name:        groupName,
+		OwnedDomain: groupOwnedDomain,
+	}
+	if err := userDB.InsertOrUpdateGroup(ctx, group); err != nil {
+		return nil, err
+	}
+
+	return &grpb.CreateGroupResponse{}, nil
```
[the remark this answers is not in the document]
Comment (bduffany) [reply server/buildbuddy_server/buildbuddy_server.go:150]: Ack, will do in a followup

**Pull request merged**